rust/src/table_state.rs

```diff
@@ -368,13 +368,14 @@ impl DeltaTableState {
         filters: &'a [PartitionFilter<'a, &'a str>],
     ) -> Result<impl Iterator<Item = &'a Add> + '_, DeltaTableError> {
         let current_metadata = self.current_metadata().ok_or(DeltaTableError::NoMetadata)?;
-        if !filters
-            .iter()
-            .all(|f| current_metadata.partition_columns.contains(&f.key.into()))
-        {
-            return Err(DeltaTableError::InvalidPartitionFilter {
-                partition_filter: format!("{filters:?}"),
-            });
+
+        for f in filters {
+            if !current_metadata.partition_columns.contains(&f.key.into()) {
+                let column = f.key.to_string();
+                return Err(DeltaTableError::ColumnNotPartitioned {
```

Comment:
nitpick, the original implementation returns all columns that aren't partitioned, but this implementation does an early return, which results in a different behavior.

Reply:
Thanks @houqp, good point! I tried to restore to the original behavior in the latest commit